src/mgxs.cpp

```diff
@@ -102,59 +102,27 @@ Mgxs::metadata_from_hdf5(hid_t xs_id, const std::vector<double>& temperature,
 
-  // If only one temperature is available, lets just use nearest temperature
-  // interpolation
-  if ((num_temps == 1) && (settings::temperature_method == TemperatureMethod::INTERPOLATION)) {
-    warning("Cross sections for " + strtrim(name) + " are only available " +
-            "at one temperature.  Reverting to the nearest temperature " +
-            "method.");
-    settings::temperature_method = TemperatureMethod::NEAREST;
-  }
-
-  switch(settings::temperature_method) {
-    case TemperatureMethod::NEAREST:
-      // Determine actual temperatures to read
-      for (const auto& T : temperature) {
-        auto i_closest = xt::argmin(xt::abs(available_temps - T))[0];
-        double temp_actual = available_temps[i_closest];
-        if (std::fabs(temp_actual - T) < settings::temperature_tolerance) {
-          if (std::find(temps_to_read.begin(), temps_to_read.end(), std::round(temp_actual))
-              == temps_to_read.end()) {
-            temps_to_read.push_back(std::round(temp_actual));
-          }
-        } else {
-          fatal_error(fmt::format(
-            "MGXS library does not contain cross sections for {} at or near {} K.",
-            in_name, std::round(T)));
-        }
+  for (int i = 0; i < temperature.size(); i++) {
+    for (int j = 0; j < num_temps; j++) {
+      if (j == (num_temps - 1)) {
+        fatal_error("MGXS Library does not contain cross sections for " +
+                    in_name + " at temperatures that bound " +
+                    std::to_string(std::round(temperature[i])));
       }
-      break;
-
-    case TemperatureMethod::INTERPOLATION:
-      for (int i = 0; i < temperature.size(); i++) {
-        for (int j = 0; j < num_temps; j++) {
-          if (j == (num_temps - 1)) {
-            fatal_error("MGXS Library does not contain cross sections for " +
-                  in_name + " at temperatures that bound " +
-                  std::to_string(std::round(temperature[i])));
-          }
-          if ((available_temps[j] <= temperature[i]) &&
-              (temperature[i] < available_temps[j + 1])) {
-            if (std::find(temps_to_read.begin(),
-                          temps_to_read.end(),
-                          std::round(available_temps[j])) == temps_to_read.end()) {
-              temps_to_read.push_back(std::round((int)available_temps[j]));
-            }
-
-            if (std::find(temps_to_read.begin(), temps_to_read.end(),
-                          std::round(available_temps[j + 1])) == temps_to_read.end()) {
-              temps_to_read.push_back(std::round((int) available_temps[j + 1]));
-            }
-            break;
-          }
+      if ((available_temps[j] <= temperature[i]) &&
+          (temperature[i] < available_temps[j + 1])) {
+        if (std::find(temps_to_read.begin(), temps_to_read.end(),
+              std::round(available_temps[j])) == temps_to_read.end()) {
+          temps_to_read.push_back(std::round((int)available_temps[j]));
```

Comment:
BTW, I think this cast to int then round thing isn't right to do, but it was already like that